esmvaltool/cmorizers/data/cmorizer.py

```diff
@@ -535,6 +535,9 @@ def _parse_date(date):
             8: "%Y%m%d",
         }
         format_string = date_formats.get(len(date_string), None)
-        if format_string is None:
-            return None
+        if len(date_string) not in [4, 6, 8]:
```

Comment:
```suggestion
        if format_string is None:
```
This way the exception throwing code here can stay the same if we add new formats in the future. That's why I used `date_formats.get` instead of `date_formats[...]`. Alternatively, you could use the latter in line 537 and wrap it in a try-except block.

Reply:
You're right, thanks! I was focusing on testing weird cases and forgot about that point.